### ISSUE TITLE
[Feature Request] Connecting trims and cables

### ISSUE THREAD

**Issue by Aproxia-dev** [tom5454/Toms-Storage#252]
I've been trying to find a mod to make my storage easier to deal with while not giving up the vanilla feel. I've stumbled upon this mod and decided to play around with it, and found myself extremely frustrated by the fact that you cannot connect cables and trims, as well as a bunch of other things that I don't really wanna go in depth with. All I'm gonna say is that I find it stupid that inventory proxies don't stack, and that the crafting recipes feel non-sensical and too expensive in many ways.

I've tried making the following storage room design
![image](https://github.com/tom5454/Toms-Storage/assets/53254254/d3256f11-168a-4ce1-b16c-fec15f4f0911)

And I've managed to find that this design perfectly connects all the chests together.
![image](https://github.com/tom5454/Toms-Storage/assets/53254254/11481b55-78ee-47f4-b325-1fc81ccf3187)

However, I ran into a problem as I've tried to stack this design as such.
![image](https://github.com/tom5454/Toms-Storage/assets/53254254/ea4c6ddb-04d4-4c99-9fd6-63a328f3cf02)

Connecting the two isles like so does not work
![image](https://github.com/tom5454/Toms-Storage/assets/53254254/49c2e14b-0fb4-4767-9145-f408ff8fb92b)

And neither does it work when I try to replace the two with cable connectors.

And I can't just use a single inventory connector, because cables don't connect with trims, even when I use a cable connector. Terminals also don't connect with trims which I find quite disappointing, as it would make routing everything so much easier. It could let me turn the initial design into quite literally just this
![image](https://github.com/tom5454/Toms-Storage/assets/53254254/16b13267-072a-4449-9c9f-0affb5d96f29)

In conclusion, I might just be coping and seething at the fact that the simple storage mod isn't as simple as I had hoped for, but like... These things feel like they should work, and yet they don't which I find extremely frustrating.

**Comment by BryceBubbles** [comment]
Hi Emily
I know this post was 2 weeks ago and you've probably moved on, but I'll comment anyway.
Btw I'm not a mod developer, just a fellow user.

Firstly, when asking for something, especially from people who don't owe you anything, perhaps leave out the "coping and seething" as you put it. It can get in the way of what may be a very reasonable request.

As for the recipes being nonsensical and expensive, I both agree and disagree. This is a vanilla style mod, which I believe doesn't add any extra "recipe only" items, and tries to be balanced and not too op. So some of the items used in the recipes do feel a little weird at first, but lots of Minecraft recipes are weird really. Compared to the possibilities and requirements of a purely vanilla storage system, this mod is extremely powerful, and some would say that these recipes are cheap, especially compared to other storage mods.

At the end of the day, mod packs will often change these recipes to fit their progression, usually making them even harder. And if you're in control of your pack (sound like you are), with a very simple datapack you could change the recipes to something that feels logical and balanced to you - if you haven't made a datapack before, it's seriously very easy. 

All that aside, there is a solution to your problem using current features. What you need is an `Inventory Proxy`.

![2023-07-01_17 43 29](https://github.com/tom5454/Toms-Storage/assets/3781435/28a9e66b-e70b-4faa-a111-cddbc6b8a97e)

Take note of which way the arrow on the Proxy is facing, you have to place a temporary block where the `Inventory Cable Connector` is, place the Proxy onto that, remove the temp block and place the Cable Connector on to the Proxy.

However, with this setup, as you remove and add items via the terminals, they wont automatically get put back in the exact specific chests that you want them in, as by default all connected inventories are just mass storage space. Not worrying about where stuff is stored is by far the cheapest way to use this mod! 

But with the storage room design you have, it looks like you'll be wanting items to go to specific chests, in which case you'd need to use `Filtered Inventory Cable Connectors` on each chest, with the appropriate filters set.

![2023-07-01_17 37 45](https://github.com/tom5454/Toms-Storage/assets/3781435/baa3d148-3837-4b89-9608-9e7c8e38cfd9)

Each filtered connector has 9 item filter slots, but by using `Item Filter` items you could filter up to 81 items. This would be quite time consuming to set up though, so I recommend using the `Polymorphic Item Filter`, filling all slots in every chest and setting `Keep last item in slot`. This will filter items based on the contents of the chests, and wont allow any slots to ever become empty end then fill with some other random item. This is similar to how some completely vanilla storage systems work, but way more compact.

![2023-07-01_17 41 36](https://github.com/tom5454/Toms-Storage/assets/3781435/69d66f0f-c920-4eae-b6bb-fb7651614840)

![2023-07-01_17 41 50](https://github.com/tom5454/Toms-Storage/assets/3781435/724482c1-4821-4349-8ad1-c5a0837eebca)

I hope that helps. 

@tom5454 All of that side, I know the tool tips explain what is possible pretty well, but is there a reason cables couldn't connect to trims as though they were the inventory connector or another cable? It's definitely a "huh?" moment when you first try doing that.

**Comment by djnever00** [comment]
Wow great discussion here, I've been messing around with this mod a bit. Haven't gotten to build my dream storage room yet but it's a work in progress. I appreciate your answer Bryce. I think the main difference is cables are gated/filtered logically, it is a higher transport strategy, hence the higher expense. Trim is dumb movement logic, connected?thenAccessible via terminal. Connecting the two is a bit of a logic trap. My first gripe with the mod was there was no slab for the trim, so I made a mod with the slab variant.
